contrib/codegen-tools/codegen/src/main/ops/org/nd4j/codegen/mixins/Mixins.kt

```diff
@@ -94,31 +94,67 @@ val scalar = Mixin("scalar"){
 val reduce = Mixin("reduce"){
     legacy = true
     Input(DataType.NUMERIC, "in") { description = "Input variable" }
+    Arg(DataType.BOOL,"keepDims"){"Whether to keep the original  dimensions or produce a shrunk array with less dimensions"}
```

Comment:
When adding new arguments, I think it makes sense to make them optional in order to keep backwards compatibility with existing code.